### PR TITLE
Fix Flysystem requiring write access

### LINKED ISSUE #2770: Flysystem has changed behaviour to not accept readonly filesystems
https://github.com/thephpleague/flysystem/commit/cfa08d0aa5ef861ed7614b2b8d9ca88465d150e0
https://github.com/thephpleague/flysystem/pull/358

```php
if (! is_writable($realRoot)) {
    throw new \LogicException('The root path '.$root.' is not writable.');
}
```

### PR DESCRIPTION
**Pull request by CarsonF**
Fixes #2770

- Subclassed Local adapter. Any write operations just return false.
- Fixed some CCP bugs
- Fixed some usages of pathogen with flysystem

### CODE REVIEW

**Comment by CarsonF** [comment]
Idk why those two tests are failing. They pass locally...

File: src/Filesystem/Local.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    protected function ensureDirectory($root)
+    {
+        if (is_dir($root) === false || !is_writable($root)) {
+            return false;
+        }
+
+        return realpath($root);
+    }
```
Comment (GawainLynch) [inline src/Filesystem/Local.php]: Changing the behaviour here doesn't make sense.

Comment (GawainLynch) [inline src/Filesystem/Local.php]: Commenting before coffee doesn't ether…  

What I meant was that if the parent directory **is** writeable, and the directory being *ensured* doesn't exist, we should create it.

File: src/Filesystem/Local.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+        // mkdir recursively creates directories.
+        // It's easier to ignore errors and check result
+        // than try to recursively check for file permissions
+        if (!is_dir($location) && !@mkdir($location, 0777, true)) {
+            return false;
+        }
+
+        return array('path' => $dirname, 'type' => 'dir');
+    }
```
Comment (GawainLynch) [inline src/Filesystem/Local.php]: I agree with @bobdenotter's reasoning for a safeguard of `777`, but I know there are lots of us that find `777` **totally** unacceptable for production systems.  Wondering if it isn't better to have a config option for this to keep the paranoid happy? :guardsman:

File: src/Filesystem/Local.php (hunk 1 above)
Comment (CarsonF) [inline src/Filesystem/Local.php]: Doh, you're right

Comment (GawainLynch) [inline src/Filesystem/Local.php]: About the coffee… Tell me about it :eyes:

File: src/Filesystem/Local.php (hunk 2 above)
Comment (CarsonF) [inline src/Filesystem/Local.php]: What we determined was:
- directories always need executable permissions (and this is only for directories)
- umask modifies the 777 (usually to make it 755).

**Comment by bobdenotter** [comment]
Merging in!

**Pull request merged**